CFC_DataCollector/trip_generator/trip_gen.py

```diff
@@ -0,0 +1,136 @@
+     
+    def make_a_to_b(self):
+        for _ in xrange(self.num_trips + self.radius*3):  ## Based on very rough estimate of how many of these end up in the ocean
+            start_index = random.randint(0, len(self.starting_points) - 1)
+            end_index = random.randint(0, len(self.ending_points) - 1)
+            starting_point = self.starting_points[start_index]
+            ending_point = self.ending_points[end_index]
+            to_add = ( starting_point, ending_point )
+            self.a_to_b.add(to_add)
+
+
+    def get_trips_from_a_to_b(self):
+        mode_tuple = ("CAR", "WALK", "BICYCLE", "TRANSIT")
+        curr_time = datetime.datetime.now()
+        curr_month = curr_time.month
+        curr_year = curr_time.year
+        curr_day = curr_time.day
+        curr_hour = curr_time.hour
+        curr_minute = curr_time.minute
+        for t in self.a_to_b:
+            mode = mode_tuple[random.randint(0, len(mode_tuple) - 1)] ## Unsophisticated mode choice
+            try:
+                otp_trip = OTP(t[0], t[1], mode, write_day(curr_month, curr_day, curr_year), write_time(curr_hour, curr_minute), True)
+                alt_trip = otp_trip.turn_into_trip(0, 0, 0)   ## ids dont matter here 
+                alt_trip.save_to_db()
+            except PathNotFoundException:
+                print "In the sea, skipping"
+                self.amount_missed += 1
+
+def geocode_address(address):
+    if address.cord is None:
+        business_geocoder = Geocoder()
+        results = business_geocoder.geocode(address.text)
+        address.cord = results
+    else:
+        results = address.cord
+    return Coordinate(results[0].coordinates[1], results[0].coordinates[0])
+
+def generate_random_locations_in_radius(address, radius, num_points):
+    # Input the desired radius in kilometers
+    print "num points is %s" % num_points
+    locations = [ ]
+    for _ in xrange(num_points):
+        loc = get_one_random_point_in_radius(address, radius)
+        locations.append(loc)
+    return locations
+
+
+def get_one_random_point_in_radius(address, radius):
+    # From https://gis.stackexchange.com/questions/25877/how-to-generate-random-locations-nearby-my-location
+    crd = geocode_address(address)
+    radius_in_degrees = kilometers_to_degrees(radius)
+    x_0 = crd.get_lon()
+    y_0 = crd.get_lat()
+    u = random.random()
+    v = random.random()
+    w = radius_in_degrees * math.sqrt(u)
+    t = 2 * math.pi * v
+    x = w * math.cos(t)
+    y = w * math.sin(t)
+    x = float(x) / float(math.cos(y_0))   # To account for Earth something 
+    return Coordinate(x + x_0, y + y_0)
+
+def test_get_random_point():
+    a = Address("2703 Hallmark Dr Belmont CA")
+    print "1 kilometer: %s" % get_one_random_point_in_radius(a, 1)
+    print "2 kms : %s" % get_one_random_point_in_radius(a, 2)
```

Comment:
I'm changing it so it actually runs a test and putting it in a test file.

On Wed, Jul 1, 2015 at 5:32 PM, shankari <notifications@github.com> wrote:

> In CFC_DataCollector/trip_generator/trip_gen.py
> <https://github.com/e-mission/e-mission-server/pull/146#discussion_r33738663>
> :
>
> > +    radius_in_degrees = kilometers_to_degrees(radius)
> > +    x_0 = crd.get_lon()
> > +    y_0 = crd.get_lat()
> > +    u = random.random()
> > +    v = random.random()
> > +    w = radius_in_degrees * math.sqrt(u)
> > +    t = 2 * math.pi * v
> > +    x = w * math.cos(t)
> > +    y = w * math.sin(t)
> > +    x = float(x) / float(math.cos(y_0))   # To account for Earth something
> > +    return Coordinate(x + x_0, y + y_0)
> > +
> > +def test_get_random_point():
> > +    a = Address("2703 Hallmark Dr Belmont CA")
> > +    print "1 kilometer: %s" % get_one_random_point_in_radius(a, 1)
> > +    print "2 kms : %s" % get_one_random_point_in_radius(a, 2)
>
> no, no, don't remove it. move it to a unit test instead.
> this is what we use unit tests for - the advantage is that then the test
> sticks around and can be run again and again to ensure that somebody else
> didn't screw things up.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/e-mission/e-mission-server/pull/146/files#r33738663>.
>